tests/ratingservice/main_test.py

```diff
@@ -90,8 +89,7 @@ def testNewRatingCalculation(self):
             ((new_rating_vote - result1_data['rating']) /
              (result1_data['votes'] + 1))
         # compare new rating vs. expected which was rounded to 4 digits after floating point
-        self.assertEqual(result2_data['rating'],
-                         math.ceil(new_rating*10000)/10000)
+        self.assertEqual(result2_data['rating'], float("%.4f" % new_rating))
```

Comment:
The test verifies calculation. String representation will not let to do it. Decimal can do but Python's json does not support the conversion. Other ways complicate implementation even more. The current implementation seems to be a less evil.
Since we need fraction number to present the rating in UI at the end it has to be smth like `float64` in frontend service.

Reply:
I'd argue it's safer to serialize as a string and write the one line to decode it back into a 4 digit decimal, rather than trying to send float values in JSON. [Floats add ambiguity](https://stackoverflow.com/questions/35709595/why-would-you-use-a-string-in-json-to-represent-a-decimal-number). It would be ideal if we could avoid all float math in the rating service, since we are working with fixed precision values

 Plus it makes the packets larger than needed, but that's not exactly a concern for our use case